### PR TITLE
Volunteer - Ahmed Essam - GSoD'21

### PR DESCRIPTION
**Pull request by Hanaffi**

### CODE REVIEW

File: jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
+---
+title: "Google Season of Docs 2021 Applicant: Souvik Biswas"
+author: Hanaffi
+categories: article
```
Comment (huan) [inline jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md]: Please use `categories: proposal` instead of `article` for a better category.

**Review by huan — CHANGES_REQUESTED** [review]
Please follow my review and make suer the CI turns green, thanks!

**Review by huan — CHANGES_REQUESTED** [review]
Please make the CI green before we can merge your great PR, thanks!

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Hanaffi Please sign the CLA (Contributors License Agreement).

**Comment by Hanaffi** [comment]
@huan Idk why it failed again :/

**Review by huan — CHANGES_REQUESTED** [review]
From <https://github.com/wechaty/wechaty.js.org/runs/2519437947> you can found the following error messages:

```
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:18:166 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:74 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:104:21 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:111 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:112 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 3]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:115:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:116:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:118 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:119 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 3]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:122:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:126:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:127:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:128:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:132:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:132:31 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:133:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:133:32 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:137:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:138:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:142:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:142:133 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:143:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:143:128 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:147:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:148:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:148:129 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:152:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:153:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:153:98 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:155:35 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:157:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:157:98 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:158:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:158:116 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:160:34 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:162:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:162:116 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:163:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:165:35 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:167:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:167:71 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:169:36 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:171:1 MD004/ul-style Unordered list style [Expected: asterisk; Actual: dash]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:171:73 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md:173 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
```

You need to fix all of them in your files and then resubmit again to see what else our CI system will tell you.

Also you can run `npm test` in your local environment to run all the tests locally.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Hanaffi Please fix these issues

Screenshot 2021-05-12 at 11 16 15 PM

**Review by huan — CHANGES_REQUESTED** [review]
Please follow the error message from CI and fix them before we can continue reviewing this PR, thanks!

**Comment by Rohitesh-Kumar-Jain** [comment]
@Hanaffi  would you like to add some more commits to this PR?

**Review by huan — CHANGES_REQUESTED** [review]
Please make the CI green before we can continue reviewing it.

P.S. You can run `npm test` locally to make sure everything is ok before `git push`

**Comment by Hanaffi** [comment]
@huan It should be working fine now. I tested it locally 💯

**Comment by Hanaffi** [comment]
@huan Its following the naming convention but IDK why the test fails.
Can we merge it?

File: jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<div align="center">
```
Comment (iamrajiv) [inline jekyll/_posts/2021-05-05-Ahmed-Essam-GSOD-Proposal.md]: @Hanaffi Please add Front Matter here. I would suggest you follow other posts in `jekyll/posts/`.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

**Review by huan — CHANGES_REQUESTED** [review]
Please make sure the PR has passed all the CI tests and turns _green_. (you can run all tests locally by run `npm test`)

**Comment by Hanaffi** [comment]
@huan  @Rohitesh-Kumar-Jain Can you check :/ ?

**Comment by Rohitesh-Kumar-Jain** [comment]
> @huan @Rohitesh-Kumar-Jain Can you check :/ ?

Nice work, now you have just one test case failing.

the test which is failing is : 

```
front matter key `author` should has a value exist in jekyll/_contributors/__VALUE__.md file
not ok 15 "jekyll/_posts/2021-05-05-gsod-2021-volunteering-proposal.md" author profile should exist at jekyll/_contributors/Ahmed Essam.md
  ---
    operator: fail
    at: Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/contributor-profile.spec.ts:40:9)
    stack: |-
      Error: "jekyll/_posts/2021-05-05-gsod-2021-volunteering-proposal.md" author profile should exist at jekyll/_contributors/Ahmed Essam.md
          at Test.assert [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:228:54)
          at Test.bound [as _assert] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.fail (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:322:10)
          at Test.bound [as fail] (/home/runner/work/wechaty.js.org/wechaty.js.org/node_modules/tape/lib/test.js:80:32)
          at Test.<anonymous> (/home/runner/work/wechaty.js.org/wechaty.js.org/tests/jekyll/contributor-profile.spec.ts:40:9)
  ...
```

This means that you need to create your contributor profile first to write a blog post. Basically, just create your contributers profile [here](https://github.com/wechaty/wechaty.js.org/tree/master/jekyll/_contributors) like other profiles.

**Comment by Hanaffi** [comment]
@Rohitesh-Kumar-Jain Thank you :)
Can you check this: https://github.com/wechaty/wechaty.js.org/pull/895 ?

**Comment by Hanaffi** [comment]
@huan @Rohitesh-Kumar-Jain Please consider this https://github.com/wechaty/wechaty.js.org/pull/898

**Pull request closed without merging**